### DISCUSSION TITLE
Re-unify DT/NC and HT/DC in selected mods leaderboard view

### DISCUSSION THREAD

**Discussion by Non-Binary-Programmer** [ppy/osu#23227] [Ideas]
In osu!stable, scores set with the Nightcore mod are visible when viewing scores set with Double Time. This makes sense, because the differences between Double Time and Nightcore are auditory changes which are generally considered to be a matter of preference rather than substance. Additionally, they have the same mod multiplier and affect performance points identically, further giving evidence that they should be considered competitively equivalent. These same arguments apply for Half Time and Daycore, although those don't have the same precedent in osu!stable (due to osu!stable not having Daycore).

For these reasons, I believe that it would be beneficial to adopt osu!stable's approach, and display Nightcore scores when viewing Double Time scores, and do the same for Half Time and Daycore. I also personally think it would be better to show Double Time scores when viewing Nightcore scores, but that is much more subjective.

**Comment by rushiiMachine** [comment]
I would go one step further and suggest unifying those mod pairs entirely. As for the Nightcore background beats, this could be added with an additional mod toggle on DT. Adding a toggle on NC to disable beats seems like it's [popular](https://github.com/ppy/osu/discussions/14209) as well, so adding the option to DT wouldn't be too far of a stretch. Legacy scores with Nightcore could be converted into DT with pitch and beat settings respectively to preserve the original replay properly. However a question would remain of how to properly handle the pitch adjustment the same way as Nightcore, since the DT pitch adjustment is not constant pitch. An option would be to use constant pitch adjustment only when the speed adjustment is 1.5x (so on by default when using DT with pitch adjustment), however I don't think this would be the best solution. Same applies for HT+DC.

While this was proposed a while [ago](https://github.com/ppy/osu/issues/9325), given that since then Daycore has been born as well as DT now having a pitch adjustment option, I wouldn't consider it unreasonable to merge these mods. NC/DC only provide unnecessary complexity at this point, especially with DT/NC leaderboards being split (afaik, cannot verify this since I don't have supporter) even though they're essentially the same mod in terms of difficulty.

**Comment by bdach** [comment]
website doesn't do this. it's a sweeping change.

it also impacts import of existing stable scores into lazer leaderboards.

both mean that this is not likely to happen anytime soon, if ever.
